manifests/params.pp

```diff
@@ -40,11 +40,9 @@
   $ssldir = "${confdir}/ssl"
   $config = "${confdir}/puppet.conf"
 
-  # The `is_pe` fact currently works by echoing out the puppet version
-  # and greping for "puppet enterprise". With Puppet 4 and PE 2015.2, there
-  # is no longer a "PE Puppet", and so that fact will no longer work.
-  # Instead check for the `is_pe` fact or if a PE provided function is available
-  $_is_pe = (getvar('::is_pe') or is_function_available('pe_compiling_server_version'))
+  # To determine if this is a PE installation or not, we check if the `pe_anchor` resource type exists
+  # it's provided by the puppet_enterprise module
+  $_is_pe = defined('pe_anchor')
```

Comment:
It looks like the test suite rely on the `is_pe` variable and removing it from the condition break the test suite.  This PR being about updating dependencies, I think we can ignore this compatibility thing for now (assuming we trust the comments), and only focus on supporting recent stdlib?

https://github.com/puppetlabs/puppetlabs-stdlib/pull/1386 might be useful to keep the current behavior (see bellow).